src/VisualStudio/Core/Test/Venus/DocumentService_IntegrationTests.vb

```diff
@@ -0,0 +1,707 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports System.ComponentModel.Composition
+Imports System.Threading
+Imports System.Windows
+Imports System.Windows.Controls
+Imports System.Windows.Media
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Classification
+Imports Microsoft.CodeAnalysis.CSharp.Syntax
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.Editor.FindUsages
+Imports Microsoft.CodeAnalysis.Editor.UnitTests
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Host
+Imports Microsoft.CodeAnalysis.Shared.Extensions
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.VisualStudio.Composition
+Imports Microsoft.VisualStudio.LanguageServices.CodeLens
+Imports Microsoft.VisualStudio.LanguageServices.FindUsages
+Imports Microsoft.VisualStudio.OLE.Interop
+Imports Microsoft.VisualStudio.Shell
+Imports Microsoft.VisualStudio.Shell.FindAllReferences
+Imports Microsoft.VisualStudio.Shell.TableControl
+Imports Microsoft.VisualStudio.Shell.TableManager
+Imports Microsoft.VisualStudio.Text
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Venus
+
+    <UseExportProvider>
+    Public Class DocumentService_IntegrationTests
+        <WpfFact, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestFindUsageIntegration() As System.Threading.Tasks.Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document FilePath="Original.cs">
+class {|Original:C|}
+{
+        [|C|]$$ c;
+}
+        </Document>
+        <Document FilePath="Mapped.cs">
+class {|Definition:C1|}
+{
+        [|C1|] c;
+}reference 
+        </Document>
+    </Project>
+</Workspace>
+
+
+            Dim exportProvider = ExportProviderCache _
+                .GetOrCreateExportProviderFactory(TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic() _
+                .WithParts(GetType(MockServiceProvider))) _
```

Comment:
Is this a DocumentServiceProvider? If so, please rename to avoid confusion with our mock implementations fo IServiceProvider.

Reply:
it is Mocked version of IServiceProvider which return FindAllReferenceWindow